### PR TITLE
Fix directory matching prefixes

### LINKED ISSUE #135: Directory rule matches only prefix not checking if its a directory
E.g. config "ee" matches also "ee-feature-pack/..." not only "ee/"

### PR DESCRIPTION
**Pull request by xstefank**
Resolves #135

### CODE REVIEW

File: src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java [hunk 1]
```diff
@@ -50,19 +50,19 @@ void testDirectoriesMentionsNewFileInDiff() throws IOException {
...
-                .when().payloadFromString(gitHubJson.jsonString())
-                .event(GHEvent.PULL_REQUEST)
-                .then().github(mocks -> {
-                    GHPullRequest mockedPR = mocks.pullRequest(gitHubJson.id());
-                    Mockito.verify(mockedPR, Mockito.times(2)).listFiles();
-                    Mockito.verify(mockedPR).comment("/cc @7125767235");
-                    Mockito.verify(mockedPR, Mockito.times(1)).listComments();
-                    Mockito.verifyNoMoreInteractions(mockedPR);
-                });
+                mocks.configFile(CONFIG_FILE_NAME).fromString(wildflyConfigFile);
```
Comment (petrberan) [inline src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java:53]: https://github.com/xstefank/wildfly-github-bot/pull/112#discussion_r1290987969

**Review by petrberan — APPROVED** [review]

File: src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java [hunk 2]
```diff
@@ -129,41 +129,73 @@ void testDirectoriesMentionsNoHitInDiff() throws IOException {
...
+        wildflyConfigFile = """
+            wildfly:
+              rules:
+                - id: "Directory Test"
+                  directories:
+                   - app
+                  notify: [7125767235]
+              format:
+                commit:
+                  enabled: false
```
Comment (petrberan) [inline src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java:157]: Does it have to be disabled? The #101 states that checks should be enabled by default

File: src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java (hunk 1 above)
Comment (xstefank) [reply src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java:53]: I know :) working on it

File: src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java (hunk 2 above)
Comment (xstefank) [reply src/test/java/io/xstefank/wildfly/bot/PRRuleDirectoryHitTest.java:157]: no, I had different issue with it, I'll remove it

**Pull request merged**